engine/table/src/main/java/io/deephaven/engine/table/impl/multijoin/typed/incopen/gen/IncrementalMultiJoinHasherChar.java

```diff
@@ -0,0 +1,345 @@
+            MultiJoinModifiedSlotTracker modifiedSlotTracker, byte trackerFlag) {
+        final CharChunk<Values> keyChunk0 = sourceKeyChunks[0].asCharChunk();
+        final int chunkSize = keyChunk0.size();
+        final LongChunk<OrderedRowKeys> rowKeyChunk = rowSequence.asRowKeyChunk();
+        for (int chunkPosition = 0; chunkPosition < chunkSize; ++chunkPosition) {
+            final char k0 = keyChunk0.get(chunkPosition);
+            final int hash = hash(k0);
+            final int firstTableLocation = hashToTableLocation(hash);
+            int tableLocation = firstTableLocation;
+            MAIN_SEARCH: while (true) {
+                int slotValue = slotToOutputRow.getUnsafe(tableLocation);
+                if (slotValue == EMPTY_OUTPUT_ROW) {
+                    final int firstAlternateTableLocation = hashToTableLocationAlternate(hash);
+                    int alternateTableLocation = firstAlternateTableLocation;
+                    while (alternateTableLocation < rehashPointer) {
+                        slotValue = alternateSlotToOutputRow.getUnsafe(alternateTableLocation);
+                        if (slotValue == EMPTY_OUTPUT_ROW) {
+                            break;
+                        } else if (eq(alternateKeySource0.getUnsafe(alternateTableLocation), k0)) {
+                            if (tableRedirSource.getLong(slotValue) != NO_RIGHT_STATE_VALUE) {
+                                throw new IllegalStateException("Duplicate key found for " + keyString(sourceKeyChunks, chunkPosition) + " in table " + tableNumber + ".");
+                            }
+                            tableRedirSource.set(slotValue, rowKeyChunk.get(chunkPosition));
+                            if (modifiedSlotTracker != null) {
+                                final int cookie = alternateModifiedTrackerCookieSource.getUnsafe(alternateTableLocation);
+                                alternateModifiedTrackerCookieSource.set(alternateTableLocation, modifiedSlotTracker.addSlot(cookie, slotValue, tableNumber, RowSequence.NULL_ROW_KEY, trackerFlag));
+                            }
+                            break MAIN_SEARCH;
+                        } else {
+                            alternateTableLocation = alternateNextTableLocation(alternateTableLocation);
+                            Assert.neq(alternateTableLocation, "alternateTableLocation", firstAlternateTableLocation, "firstAlternateTableLocation");
+                        }
+                    }
+                    numEntries++;
+                    mainKeySource0.set(tableLocation, k0);
+                    final int outputKey = numEntries - 1;
+                    slotToOutputRow.set(tableLocation, outputKey);
+                    tableRedirSource.set(outputKey, rowKeyChunk.get(chunkPosition));
+                    outputKeySources[0].set((long)outputKey, k0);
+                    mainModifiedTrackerCookieSource.set(tableLocation, EMPTY_COOKIE_SLOT);
```

Comment:
We should note that even if this output row has subsequent adds, we will ignore its modification state on its creation step.